src/Avalonia.Controls/ScrollViewer.cs

```diff
@@ -602,5 +583,38 @@ protected override void OnKeyDown(KeyEventArgs e)
                 e.Handled = true;
             }
         }
+
+        /// <summary>
+        /// Called when a change in scrolling state is detected, such as a change in scroll
+        /// position, extent, or viewport size.
+        /// </summary>
+        /// <param name="e">The event args.</param>
+        /// <remarks>
+        /// If you override this method, call `base.OnScrollChanged(ScrollChangedEventArgs)` to
+        /// ensure that this event is raised.
+        /// </remarks>
+        protected virtual void OnScrollChanged(ScrollChangedEventArgs e)
+        {
+            RaiseEvent(e);
+        }
+
+        private void OnLayoutUpdated(object sender, EventArgs e) => RaiseScrollChanged();
+
+        private void RaiseScrollChanged()
+        {
+            var extentDelta = new Vector(Extent.Width - _oldExtent.Width, Extent.Height - _oldExtent.Height);
+            var offsetDelta = Offset - _oldOffset;
+            var viewportDelta = new Vector(Viewport.Width - _oldViewport.Width, Viewport.Height - _oldViewport.Height);
+
+            if (extentDelta != default || offsetDelta != default || viewportDelta != default)
```

Comment:
I think to have correct behavior (numerically) you might want to use something like `extentDelta.IsNearlyEquals(default)` perhaps.

Reply:
added a commit to fix this.